### ISSUE TITLE
Where to acquire the data

### ISSUE THREAD

**Issue by yaboli** [Barqawiz/Shakkala#19]
Hi there,

Very impressive work you've done here! I'm interested in the data you use to train your model. What is the data you use and where did you get it? Thanks!

**Comment by linuxscout** [comment]
[Tashkeela Corpus](https://sourceforge.net/projects/tashkeela/ )
*Citation * 
```bibtex
@article{ZERROUKI2017147,
title = "Tashkeela: Novel corpus of Arabic vocalized texts, data for auto-diacritization systems",
journal = "Data in Brief",
volume = "11",
pages = "147 - 151",
year = "2017",
issn = "2352-3409",
doi = "https://doi.org/10.1016/j.dib.2017.01.011",
url = "http://www.sciencedirect.com/science/article/pii/S2352340917300112",
author = "Taha Zerrouki and Amar Balla",
keywords = "Natural language processing, Corpus, Arabic language, Diacritization",
abstract = "Arabic diacritics are often missed in Arabic scripts. This feature is a handicap for new learner to read َArabic, text to speech conversion systems, reading and semantic analysis of Arabic texts. The automatic diacritization systems are the best solution to handle this issue. But such automation needs resources as diactritized texts to train and evaluate such systems. In this paper, we describe our corpus of Arabic diacritized texts. This corpus is called Tashkeela. It can be used as a linguistic resource tool for natural language processing such as automatic diacritics systems, dis-ambiguity mechanism, features and data extraction. The corpus is freely available, it contains 75 million of fully vocalized words mainly 97 books from classical and modern Arabic language. The corpus is collected from manually vocalized texts using web crawling process."
}
```

**Comment by yaboli** [comment]
> [Tashkeela Corpus](https://sourceforge.net/projects/tashkeela/)
> *Citation *
> 
> ```
> @article{ZERROUKI2017147,
> title = "Tashkeela: Novel corpus of Arabic vocalized texts, data for auto-diacritization systems",
> journal = "Data in Brief",
> volume = "11",
> pages = "147 - 151",
> year = "2017",
> issn = "2352-3409",
> doi = "https://doi.org/10.1016/j.dib.2017.01.011",
> url = "http://www.sciencedirect.com/science/article/pii/S2352340917300112",
> author = "Taha Zerrouki and Amar Balla",
> keywords = "Natural language processing, Corpus, Arabic language, Diacritization",
> abstract = "Arabic diacritics are often missed in Arabic scripts. This feature is a handicap for new learner to read َArabic, text to speech conversion systems, reading and semantic analysis of Arabic texts. The automatic diacritization systems are the best solution to handle this issue. But such automation needs resources as diactritized texts to train and evaluate such systems. In this paper, we describe our corpus of Arabic diacritized texts. This corpus is called Tashkeela. It can be used as a linguistic resource tool for natural language processing such as automatic diacritics systems, dis-ambiguity mechanism, features and data extraction. The corpus is freely available, it contains 75 million of fully vocalized words mainly 97 books from classical and modern Arabic language. The corpus is collected from manually vocalized texts using web crawling process."
> }
> ```

Thanks for the quick response! Did you use the entire raw data? Did you apply any technique to enhance the data? Secondly, in addition to the biLSTM network you use, what else did you do to enhance model performance?

**Comment by linuxscout** [comment]
@Barqawiz

**Comment by Barqawiz** [comment]
Hello;

First thing you should know the content is huge and come from different periods.
At the beginning, we used all the data but we phased an issue of content coming from different times and taskleel level change from school to school which cause conflict to the network.
The technique to overcome this, I kept tracking the performance of batches, and printing the tashkel after each batch. then based on comparison I removed the ones that cause the network to reduce the testing performance score or led to wrong tashkel. Another thing to note most of the data are historical which caused miss match issue with modern data, so we worked with volunteers to label few more modern data to use it for testing and training, however, the amount of modern data still requires more work. There is a paper referenced in the repository homepage that compare available Tashkel neural networks, reference it for more details.
About your second question: embeddings, batch normalization and bi-directional LSTM used within the network.
